tests/db/test_unit_db_models.py

```diff
@@ -0,0 +1,667 @@
+    await db.rollback()
+
+    project_query = await db.execute(select(Project))
+    db_project = project_query.scalars().one()
+    workflow_query = await db.execute(select(Workflow))
+    db_workflow1 = workflow_query.scalars().one()
+
+    # test relationships
+    assert db_workflow1.project_id == db_project.id
+    assert db_workflow1.project == db_project
+    # test defaults
+    assert db_workflow1.task_list == []
+    assert db_workflow1.job_list == []
+
+    # using `.project_id` attribute
+    workflow2 = Workflow(name="workflow2", project_id=db_project.id)
+    db.add(workflow2)
+    await db.commit()
+    db.expunge_all()
+
+    project_query = await db.execute(select(Project))
+    db_project = project_query.scalars().one()
+
+    workflow_query = await db.execute(select(Workflow))
+    db_workflow1, db_workflow2 = workflow_query.scalars().all()
+
+    # test relationships
+    assert db_workflow1.name == workflow1.name
+    assert db_workflow2.name == workflow2.name
+    assert db_workflow2.project_id == db_project.id
+    assert db_workflow2.project == db_project
+
+
+async def test_workflows_tasks_and_workflowtasks(db):
+
+    # DB accepts totally empty WorkflowTasks
+    db.add(WorkflowTask())
+    await db.commit()
+    db.expunge_all()
+    wftask_query = await db.execute(select(WorkflowTask))
+    db_wftask = wftask_query.scalars().one()
+    # test defaults
+    assert db_wftask.task_id is None
+    assert db_wftask.meta is None
+    assert db_wftask.args is None
+    assert db_wftask.workflow_id is None
+    assert db_wftask.order is None
+    assert db_wftask.task is None
+    # delete
+    await db.delete(db_wftask)
+    wftask_query = await db.execute(select(WorkflowTask))
+    assert wftask_query.scalars().one_or_none() is None
+
+    project = Project(name="project")
+    workflow = Workflow(name="workflow", project=project)
+    tasks_common_args = dict(
+        name="name",
+        command="command",
+        input_type="input_type",
+        output_type="output_type",
+    )
+    task1 = Task(**tasks_common_args, source="source1")
+    task2 = Task(**tasks_common_args, source="source2")
+    task3 = Task(**tasks_common_args, source="source3")
+    db.add(project)
+    db.add(workflow)
+    db.add(task1)
+    db.add(task2)
+    db.add(task3)
+    await db.commit()
+    db.expunge_all()
+
+    workflow_query = await db.execute(select(Workflow))
+    db_workflow = workflow_query.scalars().one()
+    assert db_workflow.task_list == []
+    task_query = await db.execute(select(Task))
+    task_list = task_query.scalars().all()
+    assert len(task_list) == 3
+
+    for task in task_list:
+        db.add(WorkflowTask(workflow_id=db_workflow.id, task_id=task.id))
+    await db.commit()
+    db.expunge_all()
+
+    workflow_query = await db.execute(select(Workflow))
+    db_workflow = workflow_query.scalars().one()
+    for i, task in enumerate(db_workflow.task_list):
+        assert task.order == i
+
```

Comment:
Here we could add two more pieces:
1. Test insert_task method
2. Test cascade deletion (unless it's tested elsewhere)